neurobench/datasets/primate_reaching.py

```diff
@@ -52,16 +103,21 @@ def __init__(self, file_path, filename, num_steps, train_ratio=0.8,
                 split_num (int): The number of chunks to break the timeseries into. Default is 1 (no splits).
                 remove_segments_inactive (bool): Whether to remove segments longer than max_segment_length,
                                                  which represent subject inactivity. Default is False.
+                download (bool): If True, downloads the dataset from the internet and puts it in root 
+                                 directory. If dataset is already downloaded, it will not be downloaded again.
         """
         # The samples and labels of the dataset
         self.samples = None
         self.labels = None
 
         # used for input data file management
-        self.filename = filename
-        self.file_path = os.path.join(file_path, self.filename) if '.mat' in self.filename else \
-            os.path.join(file_path, self.filename + ".mat")
+        self.filename = filename if filename[-4:] == '.mat' else self.filename + ".mat"
```

Comment:
Error here with self.filename used on right side before it is init on the left side

Reply:
Fixed and please have a review!